### PR TITLE
[OPENJDK-574] Fix for Issue #262, java-default-options must check if maxMetaSpaceSize is defined

### PR DESCRIPTION
**Pull request by jfdenise**
Fix for Issue #262 , add a check for maxMetaSpaceSize

### CODE REVIEW

**Review by jerboaa — COMMENTED** [review]
Thanks for the fix. Could you please add a test similar to [this one](https://github.com/jboss-container-images/openjdk/blob/develop/tests/features/java/gc.feature#L42..L47) so that we don't regress. GHA, would verify the test for you in this PR.

**Comment by jfdenise** [comment]
@jerboaa , thank-you. I added 2 test cases to cover size only and size constrained by max.

File: tests/features/java/gc.feature [hunk 1]
```diff
@@ -53,3 +53,19 @@ Feature: Openshift OpenJDK GC tests
     Then s2i build log should contain Using MAVEN_OPTS -XX:+UseG1GC
     And container log should contain -XX:+UseG1GC
     And container log should not contain -XX:+UseParallelGC
+
+  Scenario: Check GC_METASPACE_SIZE GC configuration
+    Given s2i build https://github.com/jboss-openshift/openshift-quickstarts from undertow-servlet
+       | variable                 | value  |
+       | GC_METASPACE_SIZE    | 120    |
+    Then s2i build log should contain Using MAVEN_OPTS -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MetaspaceSize=120m
```
Comment (jerboaa) [inline tests/features/java/gc.feature]: For this test, we only care about the `-XX:MetaspaceSize=120m` value, right? I'd just add that. Same for the assertion below (i.e. container log). Finally, a `And container log should not contain integer expression expected` would be nice.

File: tests/features/java/gc.feature [hunk 2]
```diff
@@ -53,3 +53,19 @@ Feature: Openshift OpenJDK GC tests
...
+    And container log should contain -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MetaspaceSize=120m
+
+  Scenario: Check GC_METASPACE_SIZE constrained by GC_MAX_METASPACE_SIZE GC configuration
+    Given s2i build https://github.com/jboss-openshift/openshift-quickstarts from undertow-servlet
+       | variable                 | value  |
+       | GC_METASPACE_SIZE    | 120    |
+       | GC_MAX_METASPACE_SIZE    | 90    |
+    Then s2i build log should contain Using MAVEN_OPTS -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MaxMetaspaceSize=90m -XX:MetaspaceSize=90m
+    And container log should contain -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MaxMetaspaceSize=90m -XX:MetaspaceSize=90m
```
Comment (jerboaa) [inline tests/features/java/gc.feature]: Just assert `MaxMetaspaceSize` and `MetaspaceSize` values. That should be sufficient.

**Review by jerboaa — CHANGES_REQUESTED** [review]
Looks mostly good. A few comments.

File: tests/features/java/gc.feature (hunk 1 above)
Comment (jfdenise) [reply tests/features/java/gc.feature]: Thank-you. Fixed.

File: tests/features/java/gc.feature (hunk 2 above)
Comment (jfdenise) [reply tests/features/java/gc.feature]: Thank-you. Fixed.

**Review by jerboaa — APPROVED** [review]
Looks good. Thanks again!

**Comment by jerboaa** [comment]
@jmtd I'll leave merging it up to you.

**Comment by jmtd** [comment]
Filed as https://issues.redhat.com/browse/OPENJDK-574

**Pull request merged**